connect/eaas/core/deployment/extension.py

```diff
@@ -58,19 +58,28 @@ def deploy_extension(repo, client, log, tag=None):
     environment = extension['environments'][arguments.get('env', 'production')]
     env_api = client('devops').services[extension['id']].environments[environment['id']]
 
-    environment = stop_environment(environment, env_api, log)
-    if environment['status'] == 'running':
-        return
-
     try:
-        log(f'Creating variables for extension with package_id {arguments["package_id"]}.')
-        process_variables(arguments, env_api)
+        log(f'Processing variables for extension with package_id {arguments["package_id"]}.')
+        vars_updated = process_variables(arguments, env_api, log)
+
+        if (
```

Comment:
I miss an else with some log like "The version tagged with {tag} of the extension is already deployed to the {env_type} environment {env_id}."